src/main/java/com/linkedin/kmf/apps/MirrorPipelineMonitor.java

```diff
@@ -0,0 +1,448 @@
+    } else {
+      partitionsToBrokerRatio = TopicManagementServiceConfig.PARTITIONS_TO_BROKER_RATIO_DEFAULT;
+    }
+
+    List<String> zookeeperUrlsList = getList(topicManagementConfig, MirrorPipelineMonitorConfig.ZOOKEEPER_CONNECT_LIST_CONFIG);
+    if (zookeeperUrlsList == null) {
+      throw new RuntimeException(_name + ": Must specify at least one zookeeper connection.");
+    }
+
+    // Mirrored topics must have the same number of partitions but may have a different
+    // number of brokers. Separate partitionToBrokerRatios must be maintained to achieve this requirement.
+    Map<String, Integer> zookeeperBrokerCount = new HashMap<>();
+    int maxNumPartitions = 0;
+    for (String zookeeper: zookeeperUrlsList) {
+      if (zookeeperBrokerCount.containsKey(zookeeper)) {
+        throw new RuntimeException(_name + ": Duplicate zookeeper connections specified for topic management.");
+      }
+
+      int brokerCount = com.linkedin.kmf.common.Utils.getBrokerCount(zookeeper);
+      int numPartitions = (int) Math.ceil(partitionsToBrokerRatio * brokerCount);
+      maxNumPartitions = Math.max(maxNumPartitions, numPartitions);
+      zookeeperBrokerCount.put(zookeeper, brokerCount);
+    }
+
+    for (String zookeeper: zookeeperUrlsList) {
+      topicManagementConfig.put(ConsumeServiceConfig.ZOOKEEPER_CONNECT_CONFIG, zookeeper);
+      double partitionToBrokerRatio = maxNumPartitions / zookeeperBrokerCount.get(zookeeper);
+      topicManagementConfig.put(TopicManagementServiceConfig.PARTITIONS_TO_BROKER_RATIO_CONFIG, partitionToBrokerRatio);
+      _topicManagementServices.add(new TopicManagementService(topicManagementConfig, name + ":" + zookeeper));
+    }
+  }
+
+  private Map<String, Object> createProduceServiceConfig(Map<String, Object> props) {
+    Map<String, Object> produceServiceConfig = getProps(props, MirrorPipelineMonitorConfig.PRODUCE_SERVICE_CONFIG);
+
+    if (produceServiceConfig == null) {
+      throw new RuntimeException(_name + ": Must specify a produce service config.");
+    }
+
+    produceServiceConfig.put(CommonServiceConfig.TOPIC_CONFIG, props.get(CommonServiceConfig.TOPIC_CONFIG));
+
+    return produceServiceConfig;
+  }
+
+  private Map<String, Object> createConsumeServiceConfig(Map<String, Object> props) {
+    Map<String, Object> consumeConfig = getProps(props, MirrorPipelineMonitorConfig.CONSUME_SERVICE_CONFIG);
+
+    if (consumeConfig == null) {
+      throw new RuntimeException(_name + ": Must specify a consume service config.");
+    }
+
+    consumeConfig.put(CommonServiceConfig.TOPIC_CONFIG, props.get(CommonServiceConfig.TOPIC_CONFIG));
+
+    return consumeConfig;
+  }
+
+  private Map<String, Object> createTopicManagementServiceConfig(Map<String, Object> props) {
+    Map<String, Object> topicMangementConfig = getProps(props, MirrorPipelineMonitorConfig.TOPIC_MANAGEMENT_SERVICE_CONFIG);
```

Comment:
Wasn't aware of getOrDefault. I like this solution granted warnings are suppressed for unchecked casts.